webhookValidator_test.go

```diff
@@ -0,0 +1,237 @@
+/**
+ * Copyright 2021 Comcast Cable Communications Management, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package ancla
+
+import (
+	"errors"
+	"net/url"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	_ Validator = GoodURL([]ValidURLFunc{HTTPSOnlyEndpoints(), RejectHosts(nil), RejectAllIPs()})
+)
+
+func TestGoodURL(t *testing.T) {
+	tcs := []struct {
+		Desc        string
+		URL         Webhook
+		ExpectedErr error
+	}{
+		{
+			Desc: "Blank String",
+			URL: Webhook{Config: DeliveryConfig{URL: ""},
+				FailureURL: ""},
+			ExpectedErr: errInvalidURL,
+		},
+		{
+			Desc: "No https URL",
+			URL: Webhook{Config: DeliveryConfig{URL: "http://www.google.com/"},
+				FailureURL: "https://www.google.com/"},
+			ExpectedErr: errInvalidURL,
+		},
+		{
+			Desc: "Good case",
+			URL: Webhook{Config: DeliveryConfig{URL: "https://www.google.com/"},
+				FailureURL: "https://www.google.com:1030/software/index.html"},
+		},
+		{
+			Desc: "Bad Failure URL scheme",
+			URL: Webhook{Config: DeliveryConfig{URL: "https://www.google.com/"},
+				FailureURL: "127.0.0.1:1030"},
+			ExpectedErr: errInvalidFailureURL,
+		},
+		{
+			Desc: "Bad Failure URL",
+			URL: Webhook{Config: DeliveryConfig{URL: "https://www.google.com/"},
+				FailureURL: "https://127.0.0.1:1030"},
+			ExpectedErr: errInvalidFailureURL,
+		},
+		{
+			Desc: "Good case",
```

Comment:
Usually we don't want multiple test cases with the same name.  Then if one fails, we don't know which one! I would change the first good case to something like `No Alternatives Success` and then make this one `All URL Success` or something similar.  I'm not sure if we even need both though, maybe we can delete the first one?